include/DeterministicNoiseSimulator.hpp

```diff
@@ -11,30 +11,58 @@
 #include <string>
 #include <vector>
 
-class DeterministicNoiseSimulator: public Simulator {
-public:
-    //    DeterministicNoiseSimulator(std::unique_ptr<qc::QuantumComputation>& qc, const unsigned int step_number, const double step_fidelity):
-    //        qc(qc), step_number(step_number), step_fidelity(step_fidelity) {
-    //        if (step_number == 0) {
-    //            throw std::invalid_argument("step_number has to be greater than zero");
-    //        }
-    //    }
+class DeterministicNoiseSimulator: public Simulator<dd::DensityMatrixSimulatorDDPackage> {
+    using CN    = dd::ComplexNumbers;
+    using dEdge = dd::dEdge;
```

Comment:
I changed the name to `DensityMatrixPackage`. I moved the definition of `DensityMatrixPackage` to `Simulator.hpp` due to consistency (the `StochasticNoisePackage` is also defined there because of visibility issues and template magic.).

Reply:
What I actually meant here was the `dEdge` definition. In the QFR, we define `qc::VectorDD` and `qc::MatrixDD`. Something similar should probably be done for density matrices. Either `qc::DensityMatrixDD` or refactoring the above names to `dd::Vector`, `dd::Matrix` and `dd::DensityMatrix`.

Reply:
I see. I changed that.